crates/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{collections::BTreeMap, fmt};
+
+use move_core_types::resolver::{ModuleResolver, ResourceResolver};
+use move_vm_runtime::{move_vm::MoveVM, session::Session};
+use sui_cost_tables::bytecode_tables::GasStatus;
+use sui_types::{
+    base_types::{ObjectID, SuiAddress, TxContext},
+    error::ExecutionError,
+    messages::{Argument, CallArg, ObjectArg},
+    object::Owner,
+    storage::{ChildObjectResolver, ParentSync, Storage},
+};
+
+use crate::adapter::new_session;
+
+use super::types::*;
+
+pub struct ExecutionContext<
+    'v,
+    's,
+    'a,
+    'b,
+    E: fmt::Debug,
+    S: ResourceResolver<Error = E>
+        + ModuleResolver<Error = E>
+        + Storage
+        + ParentSync
+        + ChildObjectResolver,
+> {
+    vm: &'v MoveVM,
+    state_view: &'s S,
+    ctx: &'a mut TxContext,
+    gas_status: &'a mut GasStatus<'b>,
+    session: Option<Session<'s, 'v, S>>,
+    object_owner_map: BTreeMap<ObjectID, Owner>,
+    pub gas: Option<Value>,
+    pub inputs: Vec<Option<Value>>,
+    pub results: Vec<Vec<Option<Value>>>,
```

Comment:
Is it feasible to fold `inputs` into `results`?

Reply:
No... but only because `inputs` will get some more metadata in future PRs